File: app/src/main/java/com/android/test/seaworld/presenter/Presenter.java
package com.android.test.seaworld.presenter;


import com.android.test.seaworld.model.IModel;
import com.android.test.seaworld.view.IView;

public class Presenter implements IPresenter {
    private IModel model;
    private IView view;

    public Presenter(IModel model, IView view) {
        this.model = model;
        this.view = view;
        this.view.setData(model.getAnimals());
    }

    @Override
    public void oneTact() {
        model.oneDayOfWorld();

        view.refreshData();
    }

    @Override
    public void restartWorld() {
        model.resetWorldData();
        view.refreshData();
    }
}

File: app/src/main/java/com/android/test/seaworld/presenter/Presenter.kt
package com.android.test.seaworld.presenter


import com.android.test.seaworld.model.IModel
import com.android.test.seaworld.view.IView

class Presenter(private val model: IModel, private val view: IView) {

    init {
        this.view.setData(model.animals)
    }

    fun oneTact() {
        model.oneDayOfWorld()

        view.refreshData()
    }

    fun restartWorld() {
        model.resetWorldData()
        view.refreshData()
    }
}
